src/api/environment.cc

```diff
@@ -251,6 +251,23 @@ Environment* CreateEnvironment(IsolateData* isolate_data,
                                       Environment::kOwnsInspector));
   env->InitializeLibuv(per_process::v8_is_profiling);
   env->ProcessCliArgs(args, exec_args);
+  if (RunBootstrapping(env).IsEmpty()) {
+    return nullptr;
+  }
+
+  std::vector<Local<String>> parameters = {
```

Comment:
Since we are splitting `RunBootstrapping` and `StartExecution`, this requires the user not to pass any arguments that result in an async operation done in pre-execution at the moment. This is not ideal, but we may be able to split what's done in `environment.js` in a way to make sure nothing async is done during that...then the bootstrap will be divided into three phases:

1. Up to node.js: no async operation is allowed, no access to run time states
2. environment.js: no async operation is allowed, but access to run time states is allowed (warnings emitted here must be done synchronously)
3. pre_execution.js: both are allowed.